tests/forms_tests/widget_tests/test_choicewidget.py

```diff
@@ -0,0 +1,71 @@
+import copy
+
+from django.forms.widgets import ChoiceWidget
+
+from .base import WidgetTest
+
+
+class ChoiceWidgetTest(WidgetTest):
+    widget = ChoiceWidget
+
+    @property
+    def nested_widget(self):
+        return self.widget(
+            choices=(
+                ("outer1", "Outer 1"),
+                ('Group "1"', (("inner1", "Inner 1"), ("inner2", "Inner 2"))),
+            )
+        )
+
+    def test_deepcopy(self):
+        """
+        __deepcopy__() should copy all attributes properly.
+        """
+        widget = self.widget()
+        obj = copy.deepcopy(widget)
+        self.assertIsNot(widget, obj)
+        self.assertEqual(widget.choices, obj.choices)
+        self.assertIsNot(widget.choices, obj.choices)
+        self.assertEqual(widget.attrs, obj.attrs)
+        self.assertIsNot(widget.attrs, obj.attrs)
+
+    def test_options(self):
+        options = list(
+            self.widget(choices=self.beatles).options(
+                "name",
+                ["J"],
+                attrs={"class": "super"},
+            )
+        )
+        self.assertEqual(len(options), 4)
+        self.assertEqual(options[0]["name"], "name")
+        self.assertEqual(options[0]["value"], "J")
+        self.assertEqual(options[0]["label"], "John")
+        self.assertEqual(options[0]["index"], "0")
+        self.assertIs(options[0]["selected"], True)
+        # Template-related attributes
+        self.assertEqual(options[1]["name"], "name")
+        self.assertEqual(options[1]["value"], "P")
+        self.assertEqual(options[1]["label"], "Paul")
+        self.assertEqual(options[1]["index"], "1")
+        self.assertIs(options[1]["selected"], False)
+
+    def test_optgroups_integer_choices(self):
+        """The option 'value' is the same type as what's in `choices`."""
```

Comment:
This is quite inconsistent, we have `'value'` and `` `choices` ``.

Reply:
This test was moved verbatim from test_select to this new module.